### DISCUSSION TITLE
./cmi homes "There is no saved homes" should offer to make one.

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#4081] [Feature Requests]
1. When a player has no homes and uses /cmi homes or /cmi home when they do not have any homes set, a new GUI should pop up with 1 icon in a GUI that says "Set new home", they can click this and it will run /sethome.

**Comment by CloudeLecaw** [comment]
When a player klicks /home and has no homes set it will be displayed in chat. why not simply redefine the langfile text and tell him to use /sethome to set first home?

**Comment by TomLewis** [comment]
Because **people do not read**.

7 years running a server, and the BEST way to get someones attention is to pop something up in their face. If its posted in chat and you have 50+ people online, its gone, they are not watching chat like a hawk looking for messages.

Sure if you only have 5 players on this might work, IF they read.

Plus its way way more user friendly. and its very basic to implement. this isn't a big ask, don't be negative for such a simple quality of life improvement. you are also not the developer.

**Comment by CloudeLecaw** [comment]
_u see a hand down means nothing^^ but its interesting hat some ppls looking at it xS_
_i see u didnt like my opinion to this but ok^^_

> Because **people do not read**.
> 
> 7 years running a server, and the BEST way to get someones attention is to pop something up in their face. If its posted in chat and you have 50+ people online, its gone, they are not watching chat like a hawk looking for messages.
> 
> Sure if you only have 5 players on this might work, IF they read.
> 
> Plus its way way more user friendly. and its very basic to implement. this isn't a big ask, don't be negative for such a simple quality of life improvement. you are also not the developer.

im also running servers  since hmod was out in 2010 ;) and one thing that i learned is that years dont really need to mean anthing, i saw ome ppl with 3 years having collected more expearience then others with 9 or even more then i have.
Your right players often are to lazzy to read chat. that it why u need to choose other options but
your "7 years running a server" should also teach you - never force a player to something
If player has no homes its enough to tell him in a presentive way that he has no homes and use sethome - that what ALL others do too.
noone opens a belazzygui for him to force him to set home now
What u suggest is forcing a player to set home just because he has no homes and tried /home
.
.
**- trying to help start here -**
By the way you know u dont need to use chat? To POPUP it in their "face" you can use title and subtablemeasge that u also can use in the langfile. u find this info in the lanfdile itself that u can do this

here some lines form the langfile itself that tells you how you can present important lines to your users

```
 Some lines supports option to send them to custom places, like action bar, title, sub title or even create JSON/clickable messages
# If line starts with !toast! then player will get toast message (advancement popup, only 1.12 and up). Some extra variables can be used to define type and icon. example: !toast! -t:goal -icon:paper Hello world!
# If line starts with !actionbar! then player will get action bar message defined after this variable
# If line starts with !actionbar:[seconds]! then player will get action bar message for defined amount of time
# If line starts with !broadcast! then everyone will receive message. You can add extra !toast! !actionbar! or !title! to send message for everyone to specific place, in example !broadcast!!title!
# If line starts with !customtext:[cTextName]! then custom text will be taken by name provided and shown for player. In case its used after !broadcast! then everyone who is online will get this custom text message
# If line starts with !title! then player will get title message defined after this variable, in addition it can contain !subtitle! which will add subtitle message
# If line starts with !bosbar:[name]-[timer]! then player will get bossbar message defined after this variable, in addition you can define how long this message will be visible. You need to define bossbar name which can be anything you want, but lines with same name will override each other to prevent stacking
# To include clickable messages: <T>Text</T><H>Hover text</H><C>command</C><SC>Suggested text</SC>
# <T> and </T> required, other is optional
# Use /n to break line
# To have more than one JSON message use <Next>
# <C> performs command as a player who clicked
# <CC> performs command from console once
# <CCI> performs command from console every time player clicks text
# <URL> includes url
```
.
**- trying to helps ends here - stop reading hereif u allready found your help. the rest is more uninteresting things that most ppl like me have forgotten allready tomorrow and that will answered usually by noone anymore^^-**
_dont keep it to personally :-p_

by the way your right im not the developer but the fact that u opened 4 tickets with 4 suggestions and all 4 to little things releated to homes instead of one ticket where u combined your ideas will result in one thing
Your treads will be forgotten ;)
belive me im looking these threads and ppl that think its good to write 500 trhead for 500 little things since CMI started with github and also before Cmi had a github ^^  but do what you want ;)

The dev has really other things to do at the moment then looking at small wishes like this ;)
you could look in suggestesten tht where made in 2018 you will see others also has enough ideas that will be may forgotten forever ^^ specialy about home, tablist and other new features

but thats ok.
maybe next time it would be nicer to ask for a solution of better waay to inform player instead of creating a request that fast.

**Comment by broken1arrow** [comment]
Personally text middle of the screen is better and a gui (he not need to close gui and only have a text for some seconds). 

I have set up a random teleport some give a starter home (because some left the server after get to spawn 2/3 times, because they don't know better :)). They are to lazy to put down a bed or remove it after they sleep (is soo fun how some players behave).

But hey I try not force players too much. They can remove start home (if they learn the commands or read the chat) :)

**Comment by TomLewis** [comment]
@CloudeLecaw 

> by the way your right im not the developer but the fact that u opened 4 tickets with 4 suggestions and all 4 to little things releated to homes instead of one ticket where u combined your ideas will result in one thing
Your treads will be forgotten ;)

**No.** That is not how tickets work. Its 1 ticket PER idea or PER issue, so they can be easily tracked, easily closed once there done.

Also, they got implemented and are live, so you can thank me for my suggestions.

![Screenshot 2020-08-29 11 14 29](https://user-images.githubusercontent.com/144046/91634529-fb898680-e9e8-11ea-9bcc-ee1c003cc16b.png)

Please stop being passive aggressive, You are wrong and you do not listen.

Obviously different servers have different needs, and like I told you I run an active survival for 7 years, and it has players that play on it, I wrote a script to see how many players have homes and its barely any, because the message is lost in chat.

If a user **DOES NOT** have a single home set, **WHEN THEY RUN** `/sethome`  **THEN** it opens up a GUI.

You have not understood at all.

**Comment by broken1arrow** [comment]
> 
> 
> @CloudeLecaw
> 
> > by the way your right im not the developer but the fact that u opened 4 tickets with 4 suggestions and all 4 to little things releated to homes instead of one ticket where u combined your ideas will result in one thing
> > Your treads will be forgotten ;)
> 
> **No.** That is not how tickets work. Its 1 ticket PER idea or PER issue, so they can be easily tracked, easily closed once there done.
> 
> Also, they got implemented and are live, so you can thank me for my suggestions.
> 
> ![Screenshot 2020-08-29 11 14 29](https://user-images.githubusercontent.com/144046/91634529-fb898680-e9e8-11ea-9bcc-ee1c003cc16b.png)
> 
> Please stop being passive aggressive, You are wrong and you do not listen.
> 
> Obviously different servers have different needs, and like I told you I run an active survival for 7 years, and it has players that play on it, I wrote a script to see how many players have homes and its barely any, because the message is lost in chat.
> 
> If a user **DOES NOT** have a single home set, **WHEN THEY RUN** `/sethome` **THEN** it opens up a GUI.
> 
> You have not understood at all.

If you run a bigger server, you most have a gui plugin? Way not use customalias to link /sethome to a menu you make?

Or use customalias and make this command /sethome home.

Is it a big issue if player not use homes (is more players get a harder time, if they not try to learn)? Is not many custom servers some not use a plugin for homes, so can not be so new for players how to use /homes comands.

If it was your idea with gui for use icons, was good but not well thought out (i have to create new issue on this).

**Comment by TomLewis** [comment]
@broken1arrow If you run a bigger server, you most have a gui plugin?

Oh shit I actually do! deluxe menus.
It would get a big complex with logic mind, as it only needs to open a GUI if they have none, so it first needs to check they have none, as CMI already has a GUI for homes that exist already, im trying to bridge the gap between Text & the GUI that already exists.

Yes, its a learning curve, I have many other guides in place already.

What im saying is, **IF** we had the *option* in the config, to turn on opening a GUI to make a new home **IF** they player does not have any **THEN** We could trigger this with other plugins (Like putting down their first chest in GriefPreventon)

**Comment by CloudeLecaw** [comment]
a gui is still an unnessesary even if may possible way.
CMI offers other sulltions for this problem allready
Sorry may my last post wasnt so helpful (may i had a bad day^^)
So i will give you now one little sulution for this that is one way how you can release this without this extra GUI thing needed

**todo:**
- disbale home in alias.yml
- create a custom alias called home that checks for home amount ==0 execute comment 1 and 2nd command that get executed if homeamount !=0  (== - equal  != - not equal)


**placeheolder needed for this checks:**
%cmi_user_homeamount%

**affected files:**
1. alias.yml
```
  # /cmi home $1-
  /home: false
```

2. custom alias,yml
```
CustomAlias:
  home:
    Cmds:
    - check:%cmi_user_homeamount%==0! cmi sethome
    - check:%cmi_user_homeamount%==0! cmi home
    - check:%cmi_user_homeamount%!=0! cmi home
  home ?:
    Cmds:
    - check:%cmi_user_homeamount%==0! cmi sethome $1-
    - check:%cmi_user_homeamount%==0! cmi home $1-
    - check:%cmi_user_homeamount%!=0! cmi home $1-
```


**what will happen?:**
- If Player has not set a home (homemount ==0) a home is created and he gets teleported to the home
- If Player has set a home allreay (homemount !=0) he gets teleported to tge home
- In both cases if arguments are used behind (?) then it follows the same procudcure just with the arguments behind

Why does this work even if there are two ==0 checks?
It works because CMI is working like a basic line interepreter so it starts UP to DOWN like in old basic programming lanuages
.
.
_i renember u spoke of userfriendly automatism things right?
this sounds for me more userfriendly then forcing to a GUI where the player needs to create something extra_

**more infos:**
for more information about how creating custom alias i recomment
https://www.zrips.net/cmi/commands/specialized/  and https://www.zrips.net/cmi/commands/custom-alias/
and maybe for placholders  https://www.zrips.net/cmi/placeholders/
you will see there are many many ways allready in CMI where we can build wonderfull things with it :)

**Comment by TomLewis** [comment]
> 
> 
> a gui is still an unnessesary even if may possible way.
> CMI offers other sulltions for this problem allready
> Sorry may my last post wasnt so helpful (may i had a bad day^^)
> So i will give you now one little sulution for this that is one way how you can release this without this extra GUI thing needed
> 
> **todo:**
> 
>     * disbale home in alias.yml
> 
>     * create a custom alias called home that checks for home amount ==0 execute comment 1 and 2nd command that get executed if homeamount !=0  (== - equal  != - not equal)
> 
> 
> **placeheolder needed for this checks:**
> %cmi_user_homeamount%
> 
> **affected files:**
> 
>     1. alias.yml
> 
> 
> ```
>   # /cmi home $1-
>   /home: false
> ```
> 
>     1. custom alias,yml
> 
> 
> ```
> CustomAlias:
>   home
>     Cmds:
>     - check:%cmi_user_homeamount%==0! cmi sethome
>     - check:%cmi_user_homeamount%==0! cmi home
>     - check:%cmi_user_homeamount%!=0! cmi home
>   home ?
>     Cmds:
>     - check:%cmi_user_homeamount%==0! cmi sethome $1-
>     - check:%cmi_user_homeamount%==0! cmi home $1-
>     - check:%cmi_user_homeamount%!=0! cmi home $1-
> ```
> 
> **what will happen?:**
> 
>     * If Player has not set a home (homemount ==0) a home is created and he gets teleported to the home
> 
>     * If Player has set a home allreay (homemount !=0) he gets teleported to tge home
> 
>     * In both cases if arguments are used behind (?) then it follows the same procudcure just with the arguments behind
> 
> 
> Why does this work even if there are two ==0 checks?
> It works because CMI is working like a basic line interepreter so it starts UP to DOWN like in old basic programming lanuages
> .
> .
> _i renember u spoke of userfriendly automatism things right? this sounds for me more userfriendly then forcing to a GUI where the player needs to create something extra_
> 
> **more infos:**
> for more information about how creating custom alias i recomment
> https://www.zrips.net/cmi/commands/specialized/ and https://www.zrips.net/cmi/commands/custom-alias/
> and maybe for placholders https://www.zrips.net/cmi/placeholders/
> you will see there are many many ways allready in CMI where we can build wonderfull things with it :)

This is a solution! But its a bit hacky, and I'm sure not as fast as if it were just built into the plugin as its using placeholderapi extensions which do not load as soon as the server starts up.

**Comment by CloudeLecaw** [comment]
im not sure if placeholderAPI is required since this is a playerholder come from CMI itself. but usually im suing the PlaceholderAPI and never tested it without. but you could give it a try.
as far as i know the tabcomplet for this mostly needs a restart of the server.
